charmstore_internal_test.go

```diff
@@ -177,12 +177,13 @@ func (ResourceSuite) TestPublish(c *gc.C) {
 		"data": 1,
 		"lib":  2,
 	}
-	result, err := s.Publish(id, resources)
+	result, err := s.Publish(id, []string{"development"}, resources)
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(result, gc.DeepEquals, idrev)
 	f.CheckCall(c, 0, "PutWithResponse", "/"+id.Path()+"/publish", &params.PublishRequest{
```

Comment:
Unfortunately this isn't actually checking what you think it's checking.
It's running the Publish call against the old (pre-new-channels-model)
charmstore, and it's "working" because the Publish field in the request
is set, which causes the previously uploaded charm id to be made available
in its non-development form. Well, you're not even checking whether the
charm has actually been published, but if you did, it would seem to work,
but if you tried retrieving the charm with the stable channel, it would
work even though the charm has only been published to development.

Let's make this test actually test whether the charm has been published
appropriately please, and that it has *not* been published to stable.

Again, you'll need to update the charmstore dep and fix the resulting
test failures.